File: project/frontend-services/libs/device-management/mapper/src/lib/mappers/DeviceMapper.ts
import {DeviceDTO} from '@frontend-services/device-management/dto';
import {Device} from '@frontend-services/device-management/model';

export class DeviceMapper {
  static dtoToModel(dto: DeviceDTO): Device {
    return new Device(dto.id, dto.name, dto.downlink);
  }

  static modelToDto(model: Device): DeviceDTO {
    return {id: model.id, name: model.name, downlink: model.downlink};
  }
}

File: project/backend-services/identity-management-backend/application/src/main/java/pt/sensae/services/identity/management/backend/application/mapper/device/DeviceMapper.java
package pt.sensae.services.identity.management.backend.application.mapper.device;

import pt.sensae.services.identity.management.backend.application.model.device.DeviceIdDTO;
import pt.sensae.services.identity.management.backend.application.model.device.ExpelDeviceFromDomainDTO;
import pt.sensae.services.identity.management.backend.application.model.device.PlaceDeviceInDomainDTO;
import pt.sensae.services.identity.management.backend.domainservices.model.device.DeviceResult;
import pt.sensae.services.identity.management.backend.domainservices.model.device.PlaceDeviceInDomainCommand;
import pt.sensae.services.identity.management.backend.domainservices.model.device.RemoveDeviceFromDomainCommand;

public interface DeviceMapper {

    PlaceDeviceInDomainCommand dtoToCommand(PlaceDeviceInDomainDTO dto);
    
    RemoveDeviceFromDomainCommand dtoToCommand(ExpelDeviceFromDomainDTO dto);
    
    DeviceIdDTO resultToDto(DeviceResult result);
}
